sdk/core/core-xml/src/xml.ts

```diff
@@ -79,21 +55,33 @@ export function stringifyXML(obj: unknown, opts: XmlOptions = {}): string {
  * @param opts - Options that govern the parsing of given xml string
  * `includeRoot` indicates whether the root element is to be included or not in the output
  */
-export function parseXML(str: string, opts: XmlOptions = {}): Promise<any> {
-  xml2jsParserSettings.explicitRoot = !!opts.includeRoot;
-  xml2jsParserSettings.charkey = opts.xmlCharKey ?? XML_CHARKEY;
-  const xmlParser = new xml2js.Parser(xml2jsParserSettings);
-  return new Promise((resolve, reject) => {
-    if (!str) {
-      reject(new Error("Document is empty"));
-    } else {
-      xmlParser.parseString(str, (err?: Error, res?: any) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(res);
-        }
-      });
+export async function parseXML(str: string, opts: XmlOptions = {}): Promise<any> {
+  if (!str) {
+    throw new Error("Document is empty");
+  }
+
+  const validation = validate(str);
+
+  if (validation !== true) {
+    throw validation;
+  }
+
+  const parsedXml = parse(unescapeHTML(str), getParserOptions(opts));
+
+  if (!opts.includeRoot) {
+    for (const key of Object.keys(parsedXml)) {
```

Comment:
even `for` loop is used here, in practice, it would be the first and only property in `parsedXml`?

Reply:
Yeah it will bail out right away